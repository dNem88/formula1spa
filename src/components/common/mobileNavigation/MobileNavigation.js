import React, {useContext, useState, Fragment} from 'react';
import {Link} from 'react-router-dom'
import styles from './mobileNavigation.module.css'
import userContext from '../../../context/userContext'
import logo from '../../../assets/images/logos/logo.svg'
import menu from '../../../assets/images/navigation/menu.svg'
import profile from '../../../assets/images/navigation/profile.svg'
import xicon from '../../../assets/images/navigation/xicon.svg'
import ActiveMobileNavigation from '../activeMobileNavigation/ActiveMobileNavigation';

function MobileNavigation(props) {
    let context = useContext(userContext)
    const [navigation, setNavigation] = useState({active: false})
    function clickHandler(e) {
        setNavigation({active: !navigation.active})
    }

        return (
            <Fragment>
                <nav className={styles['nav']}>
                    <div onClick={clickHandler} className={styles['hamburger-icon-container']}>
                        <img src={navigation.active ? xicon : menu}/>
                    </div>
                    <div className={styles['logo-container']}>
                        <img src={logo}/>
                    </div>
                    <div className={styles['profile-icon-container']}>
                        {context.isLogged && context.user ? <Link to={'/profile'}><img src={profile}/></Link> : ''} 
                    </div>
                    {(navigation.active) ? <ActiveMobileNavigation click={clickHandler} isLogged={context.isLogged}/> : null}
                </nav>
                
            </Fragment>
        )
    
    
}

export default MobileNavigation