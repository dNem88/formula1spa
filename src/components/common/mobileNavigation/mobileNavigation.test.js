import React from 'react'
import renderer from 'react-test-renderer'
import MobileNavigation from "./MobileNavigation";
import {
    MemoryRouter
} from 'react-router-dom'
import userContext from '../../../context/userContext'

jest.mock('../../../assets/images/logos/logo.svg', () => 'logo')
jest.mock('../../../assets/images/navigation/menu.svg',  () => 'menu')
jest.mock('../../../assets/images/navigation/profile.svg', () => 'profile')
jest.mock('../../../assets/images/navigation/xicon.svg', () => 'xicon')
jest.mock('../activeMobileNavigation/ActiveMobileNavigation', () => () => 'ActiveMobile')
const mockContextGuest = {
    isLogged: false,
    user: null
}
const mockContextUser = {
    isLogged: true,
    user: 'Daniel'
}
describe('Testing MobileNavigation', () => {
    it('Render Guest Navigation', () => {
        let snapshot = renderer.create(
            <MemoryRouter>
                <userContext.Provider value={mockContextGuest}>
                    <MobileNavigation/>
                </userContext.Provider>
            </MemoryRouter>
        ).toJSON()
        expect(snapshot).toMatchSnapshot()
        expect(snapshot.type).toBe('nav')
        expect(snapshot.children.length).toBe(3)
        expect(snapshot.children[0].type).toBe('div')
        expect(snapshot.children[0].children[0].props.src).toBe('menu')
        expect(snapshot.children[1].type).toBe('div')
        expect(snapshot.children[1].children[0].type).toBe('img')
        expect(snapshot.children[1].children[0].props.src).toBe('logo')
        expect(snapshot.children[2].type).toBe('div')
        expect(snapshot.children[2].children[0]).toBeFalsy()
    })
     it('Render User Navigation', () => {
        let snapshot = renderer.create(
            <MemoryRouter>
                <userContext.Provider value={mockContextUser}>
                    <MobileNavigation/>
                </userContext.Provider>
            </MemoryRouter>
        ).toJSON()
        expect(snapshot).toMatchSnapshot()
         expect(snapshot.children[0].type).toBe('div')
         expect(snapshot.children[0].children[0].props.src).toBe('menu')
         expect(snapshot.children[1].type).toBe('div')
         expect(snapshot.children[1].children[0].type).toBe('img')
         expect(snapshot.children[1].children[0].props.src).toBe('logo')
         expect(snapshot.children[2].type).toBe('div')
         expect(snapshot.children[2].children[0].type).toBe('a')
         expect(snapshot.children[2].children[0].props.href).toBe('/profile')
         expect(snapshot.children[2].children[0].children[0].props.src).toBe('profile')
    })
})