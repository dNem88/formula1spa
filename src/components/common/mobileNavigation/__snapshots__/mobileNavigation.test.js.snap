// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MobileNavigation Render Guest Navigation 1`] = `
<nav
  className="nav"
>
  <div
    className="hamburger-icon-container"
    onClick={[Function]}
  >
    <img
      alt="menu-icon"
      src="menu"
    />
  </div>
  <div
    className="logo-container"
  >
    <img
      alt="formula1-logo"
      src="logo"
    />
  </div>
  <div
    className="profile-icon-container"
  >
    
  </div>
</nav>
`;

exports[`Testing MobileNavigation Render User Navigation 1`] = `
<nav
  className="nav"
>
  <div
    className="hamburger-icon-container"
    onClick={[Function]}
  >
    <img
      alt="menu-icon"
      src="menu"
    />
  </div>
  <div
    className="logo-container"
  >
    <img
      alt="formula1-logo"
      src="logo"
    />
  </div>
  <div
    className="profile-icon-container"
  >
    <a
      href="/profile"
      onClick={[Function]}
    >
      <img
        alt="profile-icon"
        src="profile"
      />
    </a>
  </div>
</nav>
`;
